src/main/java/org/mule/extension/ftp/internal/command/FtpCommand.java

```diff
@@ -339,7 +341,15 @@ private Optional<FTPFile> getFileFromParentDirectory(URI absoluteUri) throws IOE
 
   private Optional<FTPFile> findFileByPath(String filePath) throws IOException {
     FTPListParseEngine engine = client.initiateListParsing(filePath);
-    return findFileInEngine(engine, filePath);
+    if (engine.hasNext()) {
+      // Since it looks for a single file it should be only one file
+      FTPFile[] ftpFiles = engine.getNext(1);
+      FTPFile ftpFile = ftpFiles[0];
+      if (FilenameUtils.getName(filePath).equals(ftpFile.getName())) {
```

Comment:
I added this verification because of the `client.initiateListParsing(filePath)` finding the file without considering the encoding.